nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/work/BlobListZipWork.java

```diff
@@ -114,11 +124,35 @@ protected void updateAndCompleteStoreEntry(List<Blob> blobs) {
     @Override
     public void work() {
         openUserSession();
+        List<Blob> blobList = new ArrayList<Blob>();
+        DownloadService downloadService = Framework.getService(DownloadService.class);
+        for (String docId : docIds) {
+            DocumentRef docRef = new IdRef(docId);
+            if (!session.exists(docRef)) {
+                log.info(String.format("Cannot retrieve document '%s', probably deleted in the meanwhile", docId));
+                continue;
+            }
+            DocumentModel doc = session.getDocument(docRef);
+            Blob blob = downloadService.resolveBlob(doc);
+            if (blob == null) {
+                log.trace("Not able to resolve blob");
+                continue;
+            } else if (!downloadService.checkPermission(doc, null, blob, "download", Collections.emptyMap())) {
+                log.debug(String.format("Not allowed to bulk download blob for document %s", doc.getId()));
+                continue;
+            }
+            blobList.add(blob);
+        }
+        if (blobList.isEmpty()) {
+            log.info("No blob to be zipped");
+            updateAndCompleteStoreEntry(Collections.emptyList());
+        }
         AutomationService as = Framework.getLocalService(AutomationService.class);
         try (OperationContext oc = new OperationContext(session)) {
             Blob blob;
             if (blobList.size() == 1) {
                 blob = blobList.get(0);
+                blob.setFilename(StringUtils.isNotBlank(this.filename) ? this.filename : this.id);
```

Comment:
`StringUtils.isNotBlank(this.filename) ? this.filename : this.id` can be factorized and `this` is unnecessary.

Reply:
You could also rely on `StringUtils#defaultIfBlank`